src/sync/impl/sync_file.cpp

```diff
@@ -420,6 +426,15 @@ std::string SyncFileManager::legacy_realm_file_path(const std::string& local_use
     return path;
 }
 
+std::string SyncFileManager::legacy_local_identity_path(const std::string& local_user_identity, const std::string& realm_file_name) const
+{
+    auto escaped_file_name = validate_and_clean_path(realm_file_name);
+    std::string path_name = util::file_path_by_appending_component(user_directory(local_user_identity), escaped_file_name);
+    std::string path = path_name + c_realm_file_suffix;
+
+    return path;
+}
+
 std::string SyncFileManager::get_user_directory_path(const std::string& local_user_identity) const {
```

Comment:
maybe change the `local_user_identity` parameter name because it's not the local id anymore, same for the `user_directory` parameter?